### PR TITLE
Plans in site creation: Plan selection screen

### LINKED ISSUE #19284: Plans in Site Creation: Integrate Plan selection into Site Creation flow
## Description

Integrate Plan selection into Site Creation flow

Requirements:
1. Enable flow: Theme Selection -> Domain Selection -> **Plan Selection** -> Site Assembly -> **Checkout**
2. Use the Plan selection view developed on Calypso https://github.com/Automattic/wp-calypso/pull/82135 and display on a web view: `wordpress.com/jetpack-app-plans/`
4. If paid domain is selected, pass `paid_domain_name` parameter. Example: `paid_domain_name=domainname.com`
5. If paid plan is selected, the original URL will redirect with  `plan_id` and `plan_slug` parameters. Example: `wordpress.com/jetpack-app/plans?plan_id=1009&plan_slug=personal_bundle`
6. Add the selected paid plan to the cart (https://github.com/wordpress-mobile/WordPress-iOS/issues/21636) using `plan_id`
7. If free plan is selected, the original URL will redirect with `plan_slug=free_plan` parameter without a `plan_id`.Example: `wordpress.com/jetpack-app/plans?plan_slug=free_plan`
8. Don't include paid domain into the cart if free plan is selected
9. If paid plan + domain is added to the cart, present Checkout after Site Assembly

### PR DESCRIPTION
**Pull request by ravishanker**
Fixes #19284 

| Domains | Plans (to be updated) |
|--------|--------|
| Screenshot_20230930_121051 | Screenshot_20231015_083133|

To test:

>**Note**
> Waiting for https://github.com/Automattic/wp-calypso/pull/82135

- `Launch` JP app
- `Go` to Me-> Debug settings
- `Switch` to `Remote Feature` tab, find `plans_in_site_creation` and enable it
-  `Switch` to **My Site** tab
- `Tap` the down arrow to the right of the site name on the My Site screen
- `Tap` the + icon at the top right on the Site picker screen
- `Select` ⊕ **Create WordPress.com site** on the modal dialog
- `Tap` Skip at top right on the first step (topic screen)
- `Tap` Skip at top right on the next step (theme screen)
- `Enter` a search query (whatever you want here)
- `Verify` that result domains are as shown in the screenshot above
- `Select` a domain
- `Verify` Plans screen opens with Personal, Premium, Business, and Ecommerce plans
- `Select` a plan 
- `Verify` it return to site setup screen 
- `Verify` it opens checkout, after site setup when an annual paid plan is chosen
- `Verify` it doesn't open checkout if a free domain with free plan is chosen, and site setup is completed
- `Verify` it doesn't allow a annual plan to be selected with a free domain
- Also, `Verify`, select a paid domain on the previous step, then choose to continue with a free domain, free plan

Checkout
- Setup sandbox cookie to test checkout (see instructions on Test Plan)
- Complete checkout with a paid domain, and paid annual plan

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested various flows

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by staskus** [comment]
@ravishanker could you mention the steps to make in the code so I could test with a current Calypso build? Or maybe you have some feedback about the Calypso build to determine if we're good to merge it or if something is still missing?

**Comment by ravishanker** [comment]
> @ravishanker could you mention the steps to make in the code so I could test with a current Calypso build? Or maybe you have some feedback about the Calypso build to determine if we're good to merge it or if something is still missing?

Probably, better to test it after merging the Calypso PR.  It needs to login on Calypso, and doesn't seem to work with Calypso.live url at the moment.

**Review by irfano — APPROVED** [review]
Great job on adding plans and connecting all these steps. 👏🏻
🟡 Creating a free site and a free domain doesn't work as expected. It creates a site with a domain that was **not selected**.
🟢 I successfully purchased a domain and a plan with a fake purchase.
🟢 No network error screens in site setup and plan section displayed.
🟢 The retry button on the network error screens in the site setup and plan section worked.
🟢 Tapping the back button in various steps in the flow worked as expected.

I've addressed the 🟡 issue in another PR. Also, I haven't included my minor comments here; instead, I've added them to that PR: https://github.com/wordpress-mobile/WordPress-Android/pull/19377

**Pull request merged**
